src/components/ParentNavigationSubtitle.tsx

```diff
@@ -43,6 +61,31 @@ function ParentNavigationSubtitle({parentNavigationSubtitleData, parentReportAct
             onPress={() => {
                 const parentAction = getReportAction(parentReportID, parentReportActionID);
                 const isVisibleAction = shouldReportActionBeVisible(parentAction, parentAction?.reportActionID ?? CONST.DEFAULT_NUMBER_ID, canUserPerformWriteAction);
+
+                if (openParentReportInCurrentTab && isReportInRHP) {
+                    // If the report is displayed in RHP in Reports tab, we want to stay in the current tab after opening the parent report
+                    if (currentFullScreenRoute?.name === NAVIGATORS.SEARCH_FULLSCREEN_NAVIGATOR) {
+                        const lastRoute = currentFullScreenRoute?.state?.routes.at(-1);
+                        if (lastRoute?.name === SCREENS.SEARCH.MONEY_REQUEST_REPORT) {
+                            const moneyRequestReportID = (lastRoute?.params as SearchFullscreenNavigatorParamList[typeof SCREENS.SEARCH.MONEY_REQUEST_REPORT])?.reportID;
+                            // If the parent report is already displayed under RHP, simply dismiss the modal
+                            if (moneyRequestReportID === parentReportID) {
+                                Navigation.dismissModal();
+                                return;
+                            }
+                        }
+
+                        Navigation.navigate(ROUTES.SEARCH_MONEY_REQUEST_REPORT.getRoute({reportID: parentReportID}));
+                        return;
+                    }
+
+                    // It is the similar case as above, but in the Inbox tab
```

Comment:
I think its best if every comment can be more-less read and understood without looking up. In this case I'm not sure if this is similar case as line 67 or line 71.

I think rewrite it to be more explicit or remove it